src/app/components/ListFilter/ListFilter.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class ListFilter extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      withJS: false,
+      submitValue: '2018-01-01',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({
+      withJS: true,
+    });
+  }
+
+  handleChange(e) {
+    this.setState(
+      {
+        submitValue: e.target.value,
+      },
+      () => {
+        // this function will be replaced by submiting to endpoint
+        console.log(this.state.submitValue);
+      }
+    );
+  }
+
+  renderFieldset() {
+    const fieldsetProps = this.props.fieldsetProps;
+    const visuallyHidden = (this.state.withJS) ? 'visuallyHidden' : '';
+    const selectName = fieldsetProps.fieldsetName;
+    const selectId = `${selectName}-input`;
+    const defaultValue = (fieldsetProps.options.length && fieldsetProps.options[0].value) ?
+      fieldsetProps.options[0].value : '';
+    const optionList = (fieldsetProps.options.length) ? fieldsetProps.options.map(
+      (opt) => <option value={opt.value} key={opt.value}>{opt.name}</option>
+    ) : null;
+
+    return (
+      <fieldset>
```

Comment:
I think this fieldset should be its own component rather than being part of `ListFilter`. That way, you can reuse that component for the `date` and for the `audience` data.

Reply:
My plan is to use <ListFilter> for both `date` and `audience`. Because `date` and `audience` are independent two forms.

Reply:
I see your point. I think `date` and `audience` should be two forms with their own submit button, yet you think they should be one form together with one submit button. I think with JS it won't make difference. But without JS, one form probably better so the user won't need to click submit twice if they want to change both audience and date.

Reply:
Correct. One form will be better because you'll only have to have one submit button. That way the user can select "Spring 2016" and "Children" and get the correct form without having to submit "Spring 2016" first and then wait to submit "Children" after.

It doesn't make too much of a difference with JS but you do want to have semantic DOM elements and it's better to have two fieldsets in one form than two forms.